### PR TITLE
add item to list

### LINKED ISSUE #4: 4. As a user, I want to add a new item to my shopping list so I can start recording purchases
## Summary
Users need to be able to add items to their shopping lists. To do this, the `AddItem` view should present them with a form that allows them to enter two things:
1. The name of the item, and
2. An estimate of when they think they’ll need to buy the item again.

[The README file for the shopping list API](https://github.com/the-collab-lab/smart-shopping-list-next/blob/main/src/api/README.md) will be helpful as you work on this feature.

## Acceptance criteria
UI-related tasks:
- [x] The `Add Item` view displays a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again. There should be 3 choices for this:
	- “Soon”, corresponding to 7 days
	- “Kind of soon”, corresponding to 14 days
	- “Not soon”, corresponding to 30 days
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] When the user submits the form, they see a message indicating that the item either was or was not saved to the database.

Data-related tasks:
- [x] The `console.log` in the `addItem` function in `src/api/firebase.js`  is replaced with a function that adds the new document to the Firestore database. That function will be imported from the `firebase/firestore` module.
- [x]  The user’s soon/not soon/kind of soon choice is used to calculate `nextPurchasedDate`

## Notes
- When this feature is implemented correctly, new items will *automatically* show up in the list view because `streamListItem` gets new data from Firestore every time there’s a change.
- You will know your `<label>` is correctly implemented if you click on the label and keyboard focus moves to the related input. Refer to [the MDN docs on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for more info.
- Users will be able to submit the form with the `Enter` key if you listen for the right JavaScript event on the right element! If you're adding an `onKeyDown` listener anywhere, you might be overcomplicating things.
- The add item form has the potential to be a lot of scope. [The wireframe for the "add item" page](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/add-item.png) has several notes related to accessibility and UX.

### PR DESCRIPTION
**Pull request by fakehouseplant**
## Description

We built a form in AddItem.jsx that allows the user to submit an item name and an estimate of how soon they will repurchase this item. On submission of the form, the data is saved to the database, and the /list view is automatically updated with the new item. A message is displayed if the submission is successful or if an error occurred. Additionally in firebase.js we updated the addItem function by replacing the console.log with addDoc to store the data in the database. 

## Related Issue

closes #4 

## Acceptance Criteria

UI-related tasks:

-The Add Item view displays a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again. There should be 3 choices for this:
     “Soon”, corresponding to 7 days
     “Kind of soon”, corresponding to 14 days
     “Not soon”, corresponding to 30 days

-The input that accepts the name of the item has a semantic label element associated with it
 
-The user can submit this form with both the mouse and the Enter key
 
-When the user submits the form, they see a message indicating that the item either was or was not saved to the database.

Data-related tasks:

-The console.log in the addItem function in src/api/firebase.js is replaced with a function that adds the new document to the Firestore database. That function will be imported from the firebase/firestore module.
 
-The user’s soon/not soon/kind of soon choice is used to calculate nextPurchasedDate

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### After

Screen Shot 2023-04-14 at 8 52 29 AM

## Testing Steps / QA Criteria

I am a little unsure that the try...catch block in the onFormSubmit function at line 25 in AddItem.jsx works correctly. The success/error messages do display but possibly not in all circumstances correctly. Placing a throw 500 in the try block will force the error message to display, but when I broke the database connection intentionally, I was still receiving the success message on submit. If someone could also test this out, it would be deeply appreciated!

### CODE REVIEW

File: src/views/AddItem.jsx [hunk 1]
```diff
@@ -1,3 +1,97 @@
...
+				itemName: e.target.value,
+			}));
+		}
+	};
+
+	const onFormSubmit = (e) => {
+		e.preventDefault();
+		try {
+			addItem(listId, itemData);
+			setSubmitMessage('Item successfully added to your list!');
```
Comment (yiremorlans) [inline src/views/AddItem.jsx:29]: Very descriptive submission message.

File: src/views/AddItem.jsx [hunk 2]
```diff
@@ -1,3 +1,97 @@
...
+									checked={itemData.daysUntilNextPurchase === 30}
+								/>
+								Not Soon
+							</label>
+						</div>
+					</fieldset>
+				</div>
+				<button type="submit">Add Item</button>
+			</form>
+			<p>{submitMessage}</p>
```
Comment (yiremorlans) [inline src/views/AddItem.jsx:94]: I think the insertion of the submission message into the page is a great idea over using something like an alert.

**Review by yiremorlans — APPROVED** [review]
The code is clear and concise without the need for additional comments. I can read this and understand the expected behavior for adding new items to the list.

File: src/api/firebase.js [hunk 3]
```diff
@@ -55,7 +55,7 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	const listCollectionRef = collection(db, listId);
 	// TODO: Replace this call to console.log with the appropriate
 	// Firebase function, so this information is sent to your database!
-	return console.log(listCollectionRef, {
+	return await addDoc(listCollectionRef, {
```
Comment (adidalal) [inline src/api/firebase.js:58]: Same comment as the other PR - handling errors if addDoc fails. See https://github.com/the-collab-lab/tcl-56-smart-shopping-list/pull/18/files#diff-d274a54187c91ba0f532df2a9e194e27ab50e988f5e4c33f5a7893918320c661R35

**Pull request merged**